### ISSUE TITLE
Possible memory Leak?

### ISSUE THREAD

**Issue by PandaGGuy** [Spottedleaf/Starlight#133]
Hey there, Been trying to trackdown a memory leak on my server while pregening. I think i tracked it down to starlight?

My test server setup:
- JDK: Zulu 17
 - Java Arguments: -Xms1024M -Xmx5G -XX:MaxMetaspaceSize=512M -XX:+UnlockExperimentalVMOptions -XX:+UseShenandoahGC -XX:ShenandoahGCMode=iu -XX:ShenandoahGCHeuristics=compact -XX:ShenandoahUncommitDelay=20000 -XX:ShenandoahGarbageThreshold=10 -XX:ShenandoahAllocationThreshold=20 -XX:ShenandoahGuaranteedGCInterval=20000 -XX:ShenandoahUnloadClassesFrequency=60 -XX:MaxTenuringThreshold=1 -XX:+PerfDisableSharedMem -XX:+DisableExplicitGC -XX:+ParallelRefProcEnabled -XX:ParallelGCThreads=16 -XX:ConcGCThreads=4 -Dusing.aikars.flags=true
 - Base Mods: 
     - fabric-api-0.46.4+1.18
     -  lithium-fabric-mc1.18.1-0.7.7
     - sodium-fabric-mc1.18.1-0.4.0-alpha6+build.14
     - lazydfu-0.1.2
     - Chunky-1.2.164
     - spark-fabric
    
 Test Procedure: Spun up two servers with same exact setup from above but one with starlight and one without. Ran chunky on both servers at the same time and took a heapdump after 10 minutes. Then let both servers sit idle for 5 minutes and took another heap dump. It looks like with starlight, java is struggling to clear out memory. This eventually causes the server to either hang/ crash due to out of memory/ or another occasional random crash while pregenning.
 
 
 Example occasional crash while pregening w/ chunky on starlight server
![image](https://user-images.githubusercontent.com/11316863/153774973-246fa76e-6a06-4e49-8b2c-357a6e5dcb0b.png)

It looks like once pregening is done, the server is fine and memory is able to clear out but while pregening it keeps building which causes issues when Pregening large modded worlds.

Please find heapdumps in dropbox folder below:
[Speak Heap Dump Folder](https://www.dropbox.com/sh/6igw57smug1dpxo/AACH8dHMDCTa8CksqOjcQ_yea?dl=0)

Heads up, I do not have much dev experience in java. So if this was misidentified, please let me know!

**Comment by RadiantSol** [comment]
Can confirm I was having the same issue when generating with Chunky. After testing by uninstalling each mod one by one, the memory leak issues disappeared when I removed Starlight and restarted the pregeneration.

**Comment by ryleu** [comment]
I've run into the same thing, but only with chunky and not with other pregeneration mods. It's probably an issue on starlight's end.

**Comment by CrazyWolf13** [comment]
I can confirm Chunky with starlight 0.2 CPS without starlight 12 CPS. 
ETA for some chunk generation with starlight 400 hours without starlight 16 hours.

**Comment by ghost** [comment]
Interesting to see Starlight's `SWMRNibbleArray[]` appear as one of the larger heap users:
![image](https://user-images.githubusercontent.com/24266948/180128799-0ace1840-0bd1-4a86-8dd6-0d6ff1c05945.png)

I'm not even pre-generating large areas (I'm just flying around in the world!) and it is retaining 148MB of heap space.

**Comment by Funtimes909** [comment]
Can confirm I am also having this issue

**Comment by Spottedleaf** [comment]
The light engine can outpace the chunk system on these versions especially unfortunately. 

The light arrays are only referenced by the LevelChunk, so the chunk system is not unloading chunks which is exactly what's happening in https://github.com/PaperMC/Starlight/issues/88

Duplicate of https://github.com/PaperMC/Starlight/issues/88
